drs_cli/client.py

```diff
@@ -1,195 +1,428 @@
+                )
+            logger.info(f"Object deleted: {object_id}")
+        return response_val
+
+    def _get_headers(self) -> Dict:
+        """Build dictionary of request headers.
 
-    def _get_headers(self):
-        """Set dictionary of request headers in `self.headers`."""
-        self.headers: Dict = {
+        Returns:
+            A dictionary of request headers
+        """
+        headers: Dict = {
             'Content-type': 'application/json',
         }
         if self.token:
-            self.headers['Authorization'] = 'Bearer ' + self.token
+            headers['Authorization'] = 'Bearer ' + self.token
+        return headers
+
+    def _get_host(
+        self,
+        uri: str,
+    ) -> Tuple[str, str]:
+        """Extract URI schema and domain or IP from HTTP, HTTPS or DRS URI.
+
+        Arguments:
+            uri: HTTP or HTTPS URI pointing to the root domain/IP of a DRS
+                instance OR a hostname-based DRS URI to a given object, cf.
+                https://ga4gh.github.io/data-repository-service-schemas/preview/develop/docs/#_hostname_based_drs_uris.
+                Anything after a slash following the domain/IP will be ignored.
+
+        Returns:
+            Tuple of URI schema (e.g., 'https', 'drs') and host domain or IP
+            (e.g., 'my-drs.app', '0.0.0.0').
+
+        Raises:
+            drs_cli.errors.InvalidURI: input URI cannot be parsed.
+
+        Examples:
+           >>> DRSClient.get_host(uri="https://my-drs.app/will-be-ignored")
+           ('https', 'my-drs.app')
+           >>> DRSClient.get_host(uri="drs://my-drs.app/My0bj3ct")
+           ('drs', 'my-drs.app')
+        """
+        match = re.search(self._RE_HOST, uri, re.I)
+        if match:
+            schema = match.group('schema')
+            host = match.group('host').rstrip('\\')
+            if len(host) > 253:
+                raise InvalidURI
+            return (schema, host)
+        else:
+            raise InvalidURI
+
+    def _get_object_id(
+        self,
+        object_id: str,
+    ) -> str:
+        """
+        Arguments:
+            object_id: Implementation-specific DRS identifier OR hostname-based
+               DRS URI pointing to a given object, cf.
+               https://ga4gh.github.io/data-repository-service-schemas/preview/develop/docs/#_hostname_based_drs_uris.
+               Note that if a DRS URI is passed, only the DRS identifier part
+               will be evaluated. To reset the hostname, create a new client
+               with the `DRSClient()` constructor.
+
+        Returns:
+            Validated, percent-encoded object ID.
+
+        Raises:
+            drs_cli.errors.InvalidObjectIdentifier: input object ID cannot be
+                parsed.
+        """
+        match = re.search(self._RE_OBJECT_ID, object_id, re.I)
+        try:
+            return quote(string=match.group('obj_id'), safe='')
+        except Exception:
+            raise InvalidObjectIdentifier
```

Comment:
This single line is only left from achieving `100%` coverage. I tried several test cases, but not able to make `quote` raise an error. :/